src/main/scala/sbtprotoc/ProtocPlugin.scala

```diff
@@ -190,14 +188,17 @@ object ProtocPlugin extends AutoPlugin {
         def downloadArtifact(moduleId: ModuleID): Future[File] =
           Future {
             blocking {
-              IO.withTemporaryDirectory { tmpDir =>
-                lm.retrieve(
-                  moduleId,
-                  None,
-                  tmpDir,
-                  log
-                ).fold(w => throw w.resolveException, _.head)
-              }
+              IO.withTemporaryDirectory(
+                { tmpDir =>
+                  lm.retrieve(
+                    moduleId,
+                    None,
+                    tmpDir,
```

Comment:
`retrieveDirectory` seems to be ignored when LM is backed by Coursier - a permanent coursier path is always returned. That's why tests are currently passing despite the current default `keepDirectory = false`.

Reply:
Sounds like sbt 1.2.x actually respects the download directory which is automatically wiped before the downloaded executable is copied out of it? I would still like to have the temp directory automatically deleted. If my understanding is correct, the scope of `IO.withTemporaryDirectory` is incorrect.

Reply:
> Sounds like sbt 1.2.x actually respects the download directory which is automatically wiped before the downloaded executable is copied out of it?

Exactly. sbt 1.2.x and I believe later versions with coursier disabled.

>  I would still like to have the temp directory automatically deleted. If my understanding is correct, the scope of IO.withTemporaryDirectory is incorrect.

Looking at the current design of `FileCache`, I don't think that's possible. The copy operation could be exposed to the client by providing a callback?